tests/integration/test_oidc_client.py

```diff
@@ -0,0 +1,47 @@
+from urllib.parse import quote
+
+from flask import url_for
+import pytest
+import requests
+
+
+class TestOIDCClient(object):
+
+    def test_authenticate(self, provider, client):
+        assert provider.openid_configuration.called
+
+        response = client.get(url_for('broker.auth'))
+        assert response.status_code == 302
+
+        response = requests.get(response.location, allow_redirects=False)
+        assert provider.authorization_endpoint.called
+
+        response = client.get(response.headers['location'])
+        assert provider.token_endpoint.called
+        assert provider.userinfo_endpoint.called
+
+    def test_idp_hint_passed(self, provider, client):
```

Comment:
Need to update class to start with it_ rather than Test as pytest.ini updated to BDD style